### PR TITLE
Minimal APIs and native AOT /4

### LINKED ISSUE #29915: Minimal APIs and native AOT
[Minimal APIs and native AOT](https://devblogs.microsoft.com/dotnet/asp-net-core-updates-in-dotnet-8-preview-3/#minimal-apis-and-native-aot)

---
#### Document Details

⚠ *Do not edit this section. It is required for learn.microsoft.com ➟ GitHub issue linking.*

* ID: 8839fa1e-7c10-4b22-d03d-669906a2b9ea
* Version Independent ID: 8839fa1e-7c10-4b22-d03d-669906a2b9ea
* Content: [ASP.NET Core support for native AOT](https://learn.microsoft.com/en-us/aspnet/core/fundamentals/native-aot?view=aspnetcore-8.0&viewFallbackFrom=aspnetcore-7.0)
* Content Source: [aspnetcore/fundamentals/native-aot.md](https://github.com/dotnet/AspNetCore.Docs/blob/main/aspnetcore/fundamentals/native-aot.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-fundamentals**
* GitHub Login: @mitchdenny
* Microsoft Alias: **midenn**

---
[Associated WorkItem - 141714](https://dev.azure.com/msft-skilling/Content/_workitems/edit/141714)

### PR DESCRIPTION
**Pull request by Rick-Anderson**
Fixes #29915

[Internal review URL](https://review.learn.microsoft.com/en-us/aspnet/core/fundamentals/native-aot?view=aspnetcore-8.0&branch=pr-en-us-29967#minimal-apis-and-json-payloads)

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [aspnetcore/fundamentals/native-aot.md](https://github.com/dotnet/AspNetCore.Docs/blob/ba5fa32465a28ad330f21a317d5fa208fbbf4c9f/aspnetcore/fundamentals/native-aot.md) | [ASP.NET Core support for native AOT](https://review.learn.microsoft.com/en-us/aspnet/core/fundamentals/native-aot?branch=pr-en-us-29967) |

### CODE REVIEW

File: aspnetcore/fundamentals/native-aot.md [hunk 1]
```diff
@@ -238,6 +238,22 @@ The `dotnet publish` command also compiles the source files and generates files
...
+* Requires the use of the [`System.Text.Json` source generator](/dotnet/standard/serialization/system-text-json/source-generation).
+
+All types accepted as parameters to or returned from request delegates in Minimal APIs apps must be configured on a <xref:System.Text.Json.Serialization.JsonSerializerContext> that is registered via ASP.NET Core’s dependency injection:
+
+:::code language="csharp" source="~/fundamentals/aot/samples/Program.cs" highlight="7-10,25-99":::
+
+In the preceding highlighted code:
+
+* The JSON serializer context is registered with the [DI container](xref:fundamentals/dependency-injection).
+* The custom `JsonSerializerContext` is annotated with the [`[JsonSerializable]`](/dotnet/api/system.text.json.serialization.jsonserializableattribute) attribute to enable source generated JSON serializer code for the `ToDo` type.
```
Comment (Rick-Anderson) [inline aspnetcore/fundamentals/native-aot.md:255]: @BrennanConroy please review.

Generated template code:
```
namespace MyFirstAotWebApi;

public class Todo
{
    public int Id { get; set; }
    public string? Title { get; set; }
    public DateOnly? DueBy { get; set; }
    public bool IsComplete { get; set; }
}

static class TodoGenerator
{
    private static readonly (string[] Prefixes, string[] Suffixes)[] _parts = new[]
        {
            (new[] { "Walk the", "Feed the" }, new[] { "dog", "cat", "goat" }),

...
```

File: aspnetcore/fundamentals/native-aot.md [hunk 2]
```diff
@@ -238,6 +238,22 @@ The `dotnet publish` command also compiles the source files and generates files
...
 [!INCLUDE[](~/fundamentals/aot/includes/aot_lib.md)]
 
+## Minimal APIs and JSON payloads
+
+The Minimal API framework is optimized for receiving and returning JSON payloads using <xref:System.Text.Json?displayProperty=fullName>. `System.Text.Json`:
+
+* Imposes compatibility requirements for JSON and native AOT.
+* Requires the use of the [`System.Text.Json` source generator](/dotnet/standard/serialization/system-text-json/source-generation).
+
+All types accepted as parameters to or returned from request delegates in Minimal APIs apps must be configured on a <xref:System.Text.Json.Serialization.JsonSerializerContext> that is registered via ASP.NET Core’s dependency injection:
```
Comment (mitchdenny) [inline aspnetcore/fundamentals/native-aot.md:248]: "All types that are transmitted as part of the HTTP body..."

You can have a parameter on the delegate that isn't bound to the body and does not to be serializable. For example, a query string parameter which is a rich object type which implements `IParsable<T>`.

**Review by mitchdenny — COMMENTED** [review]
Just a minor clarification is required. Including the Todo as suggested by Brennan is a good add as well for reference.

**Comment by Rick-Anderson** [comment]
> Just a minor clarification is required. Including the Todo as suggested by Brennan is a good add as well for reference.

@mitchdenny  I think I addressed all you and @BrennanConroy  concerns.

**Review by mitchdenny — APPROVED** [review]

**Pull request merged**